Core/HW/MediaEngine.cpp

```diff
@@ -499,7 +510,7 @@ bool MediaEngine::setVideoStream(int streamNum, bool force) {
 
 		AVStream *stream = m_pFormatCtx->streams[streamNum];
 #if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(57, 33, 100)
-		AVCodec *pCodec = avcodec_find_decoder(stream->codecpar->codec_id);
+		const AVCodec *pCodec = avcodec_find_decoder(stream->codecpar->codec_id);
```

Comment:
When `#define AVCodec const AVCodec` get expanded how does this even compile? Is double const legal in some compiler? I'm quite confused here tbh...

Reply:
Yeah I was saying they should match, which means adjust the values to what they should be. And I don't feel like being the one to test that they're right...

Double/triple/etc const is only legal if the consts are about different parts of the type, like a `const char *const my_var;` and means both that the pointer and the data pointed to is const. Not sure if that applies in this case, didn't look very closely.

Reply:
This should be fixed in the latest push.

Reply:
As for the version, `56.58.58` is ffmpeg 4.4.4, `57.59.59` is 5.1.4 and `58.60.60` is 6.0 (and remains with 6.1.1). I don't know which one should be used though I would guess 57.59.59 would be better? It's the last version of ffmpeg 5.x. https://ffmpeg.org/releases/?C=N;O=D

Reply:
I determined the correct version of libavformat is `59.16.100`. This is ffmpeg 5.0, when functions accepting `AVCodec*` added `const`.